textile/chat-features.textile

```diff
@@ -289,6 +298,16 @@ Each chat room can be configured individually, allowing options to be passed as
 ** @(CHA-RC4a)@ @[Testable]@ Where room options have not been provided, the client shall provide defaults.
 ** @(CHA-RC4b)@ @[Testable]@ Where a partial room options have been provided, the client shall deep-merge the provided values with the defaults.
 
+h2(#roos-disposal). Rooms Disposal
+
+Rooms in the Chat SDK are a complex type that register many different listeners and consume many different events. Therefore, there must be methods available to clean up any listeners to allow for garbage collection.
+
+** @(CHA-RD1)@ @[Testable]@ The @Rooms@ map / manager must offer an internal @dispose@ method that must prepare the rooms manager for garbage collection and release all managed rooms.
+*** @(CHA-RD1a)@ @[Testable]@ When @dispose@ is called, it must mark the instance as disposed to prevent future room operations.
+*** @(CHA-RD1b)@ @[Testable]@ If no rooms exist when @dispose@ is called, the method must complete successfully without error.
+*** @(CHA-RD1c)@ @[Testable]@ If rooms exist in the map, the @dispose@ method must release all rooms currently in the rooms map concurrently.
+*** @(CHA-RD1d)@ @[Testable]@ The @dispose@ method must wait for all room release operations to complete before resolving.
+
```

Comment:
I thought the point of the spec generally is public API. These points describe how things work internally. Do we want these in the spec?

Reply:
Whilst we've tried to focus it on the public API where possible, there are occasions with complex internal behaviour that needs specifying to avoid implementation divergence (e.g. RTL3d specifies performing an internal attach operation where necessary). We did the same with the room lifecycle manager.